casper/src/main/rholang/BasicWallet.rho

```diff
@@ -0,0 +1,65 @@
+//scalapackage coop.rchain.rholang.wallet
+
+//assumes standard purse API from MakeMint
+contract @"BasicWallet"(@purse, @algorithm, @pk, return) = {
+  new cryptoVerifyCh in {
+    match algorithm {
+      "ed25519" => {
+        cryptoVerifyCh!(["ed25519Verify"])
+      }
+      _ => {
+        cryptoVerifyCh!([])
+      }
+    } |
+    for(@maybeCryptoVerify <- cryptoVerifyCh) {
+      match maybeCryptoVerify {
```

Comment:
isn't this a second match on the same condition?  

if at line 11 you just ```return ! ([])``` this match is not needed

Reply:
That's a good point, though it would then also leave the dangling `for(@cryptoVerify <- cryptoVerifyCh)`. Which if fine when we have a garbage collector, but I don't know when we plan on having that.